### PR TITLE
Auto retry failed tests up to 3 times

### PR DESCRIPTION
**Pull request by taquitos**
Set the test plans to auto retry failed tests up to 3 times and use test plans with scan.

resolves [csdk-339]

### CODE REVIEW

**Review by NachoSoto — COMMENTED** [review]
This makes a lot of sense for CI, but I feel like this isn't great for developing locally. It would take longer to find out something broke, or hide flaky tests.

Maybe we should have a separate test plan for CI?

**Comment by NachoSoto** [comment]
We could reuse the "Coverage" one. I created that one for CI.

**Comment by taquitos** [comment]
Ohhh duh, yeah.

**Comment by taquitos** [comment]
> We could reuse the "Coverage" one. I created that one for CI.

I decided to create new ones, that felt a lot cleaner to me. That way we can still have a separate case for coverage.

**Comment by taquitos** [comment]
One neat benefit to this PR is that it cuts our CI time from 11 min to 9 min.
I think this is mostly due to moving to Xcode 13.4.1

File: Tests/TestPlans/RevenueCat.xctestplan [hunk 1]
```diff
@@ -10,7 +10,8 @@
   ],
   "defaultOptions" : {
     "codeCoverage" : false,
-    "testExecutionOrdering" : "random"
+    "testExecutionOrdering" : "random",
+    "testRepetitionMode" : "retryOnFailure"
```
Comment (NachoSoto) [inline Tests/TestPlans/RevenueCat.xctestplan:14]: Did you mean to leave this too? I use this test plan when I don't wanna run the SK tests

**Review by NachoSoto — COMMENTED** [review]
👍🏻 just one last question

File: Tests/TestPlans/RevenueCat.xctestplan (hunk 1 above)
Comment (taquitos) [reply Tests/TestPlans/RevenueCat.xctestplan:14]: Ugh. Sorry. Me rn

image

Comment (taquitos) [reply Tests/TestPlans/RevenueCat.xctestplan]: Fixed 🤦‍♂️

**Review by NachoSoto — APPROVED** [review]

**Pull request merged**